core/src/navigation/services/semi-collapsed-navigation.js

```diff
@@ -7,11 +7,24 @@ class SemiCollapsibleNavigationClass {
     this.responsiveNavSetting = LuigiConfig.getConfigValue(
       'settings.responsiveNavigation'
     );
-    this.semiCollapsible =
+    let currentCollapsedState = JSON.parse(
+      localStorage.getItem(NavigationHelpers.COL_NAV_KEY)
+    );
+    const isResponsiveNavSemiCollapsibleOrFiori3 =
       this.responsiveNavSetting === 'semiCollapsible' ||
-      this.responsiveNavSetting === 'Fiori3'
-        ? true
-        : false;
+      this.responsiveNavSetting === 'Fiori3';
+
+    //checking if here was a previous state in localStorage before the reload
+    if (
+      currentCollapsedState != false &&
+      isResponsiveNavSemiCollapsibleOrFiori3
+    ) {
+      this.isSemiCollapsed = this.getCollapsed();
+    }
+    this.semiCollapsible = isResponsiveNavSemiCollapsibleOrFiori3
+      ? true
+      : false;
```

Comment:
```suggestion
    this.semiCollapsible = isResponsiveNavSemiCollapsibleOrFiori3;
```
Also, since `semiCollapsible` is same as isResponsiveNavSemiCollapsibleOrFiori3. Please check do we need to keep `semiCollapsible` variable?

Reply:
thanks for double-checking, We need it for the mobile collapsed state as well, so that on mobile the left side nav always set to 'true' even after the reload.  And setting `this.semiCollapsible = isResponsiveNavSemiCollapsibleOrFiori3;` doesn't give a chance to make it `false`. 

So I'd rather leave it as it is. What do you think?